filesmanager/tests/test_tasks.py

```diff
@@ -0,0 +1,243 @@
+                        {
+                            "id": "file",
+                            "parentId": "folder",
+                            "name": "file",
+                            "type": "file",
+                            "isDir": False,
+                            "path": "Root/folder/file",
+                            "metadata": {
+                                "id": "asset_key",
+                                "asset_key": "asset_key",
+                                "display_name": "file.pdf",
+                                "url": "http://localhost:8000/media/asset_key.pdf",
+                                "content_type": "application/pdf",
+                                "file_size": "123123",
+                                "external_url": "http://localhost:8000/media/asset_key.pdf",
+                                "thumbnail": "http://localhost:8000/media/asset_key.pdf",
+                            },
+                        },
+                        {
+                            "id": "file2",
+                            "parentId": "folder",
+                            "name": "file2",
+                            "type": "file",
+                            "isDir": False,
+                            "path": "Root/folder/file2",
+                            "metadata": {
+                                "id": "asset_key2",
+                                "asset_key": None,
+                                "display_name": "file2.pdf",
+                                "url": "http://localhost:8000/media/asset_key2.pdf",
+                                "content_type": "application/pdf",
+                                "file_size": "123123",
+                                "external_url": "http://localhost:8000/media/asset_key2.pdf",
+                                "thumbnail": "http://localhost:8000/media/asset_key2.pdf",
+                            },
+                        }
+                    ],
+                }
+            ],
+        }
+
+    @patch('filesmanager.filesmanager.create_zip_file_task')
+    def test_xblock_calls_download_zip_file(self, mock_create_zip_file_task):
+        """
+        Test that the xblock calls download_zip_file with the correct arguments.
+        """
+        data = {
+            "contents": self.xblock.directories["children"][1],
+        }
+        request = Mock()
+        request.method = 'POST'
+        request.body = json.dumps(data).encode('utf-8')
+        mock_task_result = Mock()
+        mock_task_result.id = 'task_id'
+        mock_create_zip_file_task.delay.return_value = mock_task_result
+
+        self.xblock.download_content(request)
+
+        mock_create_zip_file_task.delay.assert_called_once_with(data["contents"])
+
+    def test_xblock_download_content_without_contents(self):
+        """
+        Test that the xblock returns an error when the request doesn't have the contents.
+        """
+        data = {}
+
```

Comment:
```suggestion
```

Reply:
If we're following the https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/ AAA standard, when the request setup is still part of the arrange. Try applying this to the rest of the rests